lib/src/model/broadcast/broadcast_round_controller.dart

```diff
@@ -22,22 +22,42 @@ class BroadcastRoundController extends _$BroadcastRoundController {
 
   StreamSubscription<SocketEvent>? _subscription;
 
+  Timer? _timer;
+
   late SocketClient _socketClient;
 
   @override
   Future<BroadcastRoundGames> build(BroadcastRoundId broadcastRoundId) async {
     _socketClient = ref
-        .read(socketPoolProvider)
+        .watch(socketPoolProvider)
         .open(BroadcastRoundController.broadcastSocketUri(broadcastRoundId));
 
     _subscription = _socketClient.stream.listen(_handleSocketEvent);
 
     ref.onDispose(() {
       _subscription?.cancel();
+      _timer?.cancel();
     });
 
-    return await ref.withClient(
-      (client) => BroadcastRepository(client).getRound(broadcastRoundId),
+    final games =
+        await ref.watch(broadcastRoundProvider(broadcastRoundId).future);
+
+    _timer = Timer.periodic(
+      const Duration(seconds: 1),
+      (_) => _updateClocks(),
+    );
+
+    return games;
+  }
+
+  void _updateClocks() {
+    state = AsyncData(
+      state.requireValue.map((gameId, game) {
+        if (!game.isPlaying) return MapEntry(gameId, game);
+        final thinkTime = game.thinkTime;
+        final newThinkTime = thinkTime + const Duration(seconds: 1);
```

Comment:
Why is `Timer.periodic not precise ?

The doc says
> No more than n callbacks will be made in `duration * n` time, but the time between two consecutive callbacks can be shorter and longer than duration.

I understand it as it is accurate at the second which is what is needed for broadcast.

Reply:
The time between 2 consecutive callbacks is not precisely one second as explained:
>  the time between two consecutive callbacks can be shorter and longer than duration

So if you remove a second each time when not necessarily a second has elapsed, it is not really precise. That's why we never implement clocks like that.

Reply:
I'd like the doc to state that more than n-1 callbacks will be made in duration * n time but that's obviously not the case. With only what they say in the doc I agree that there is zero guarantee on the accuracy.

I'll try to use the `Stopwatch` but it might be a bit more complicated because it has to be shared with all boards with the current structure. I need to think more about it.

Reply:
That is less important than the other comments I made (on the code architecture). So you can ignore it for now.

Reply:
I think that I fixed the code architecture now. What should I do with the clocks ?

I merged main and tried to use the new `CountdownClockBuilder` widget but I still have the issue of sharing clocks between two screens.

Reply:
What issue? I think you can maintain a `Map<BroadcastGameId, DateTime>` in the controller, and then pass the `clockUpdateAt` date time to `CountdownClockBuilder`.

Reply:
Oh right I didn't think of using the `updetedAt` field like this. Thanks.

Reply:
I had to change the `didUpdateWidget` method of the `CountdownClockBuilder` because it wasn't changing the `timeLeft` when switching between inactive clocks which is the case when going through moves on the broacast game screen.